File: src/main/java/com/cout970/magneticraft/util/Log.java
package com.cout970.magneticraft.util;

import net.darkaqua.blacksmith.log.ILog;
import net.darkaqua.blacksmith.log.LogFactory;

/**
 * Created by cout970 on 06/12/2015.
 */
public class Log {

    private static ILog LOGGER;

    static {
        LOGGER = LogFactory.createLog("Magneticraft");
    }

    public static void error(String s) {
        LOGGER.error(s);
    }

    public static void debug(Object s) {
        LOGGER.info(String.valueOf(s));
    }

    public static void info(String s) {
        LOGGER.info(s);
    }

    public static void warn(String s) {
        LOGGER.warn(s);
    }
}

File: src/main/kotlin/com/cout970/magneticraft/util/Log.kt
package com.cout970.magneticraft.util

import org.apache.logging.log4j.Logger

/**
 * Created by cout970 on 11/06/2016.
 */
object Log {

    lateinit var log: Logger

    fun setLogger(l: Logger) {
        log = l
    }

    fun info(str: String) {
        log.info("[$MODID]$str")
    }

    fun debug(vararg obj: Any?) {
        val s = StringBuilder()
        var i = obj.size
        for (o in obj) {
            s.append(o)
            i--
            if (i > 0) {
                s.append(", ")
            }
        }
        log.info("[$MODID][DEBUG]${s.toString()}")
    }
}